axelor-business-project/src/main/java/com/axelor/apps/businessproject/web/InvoiceController.java

```diff
@@ -322,10 +321,7 @@ public void exportAnnex(ActionRequest request, ActionResponse response) throws A
   public void updateLines(ActionRequest request, ActionResponse response) throws AxelorException {
     Invoice invoice = request.getContext().asType(Invoice.class);
     invoice = Beans.get(InvoiceRepository.class).find(invoice.getId());
-
-    for (InvoiceLine invoiceLine : invoice.getInvoiceLineList()) {
-      invoiceLine.setProject(invoice.getProject());
-    }
+    invoice = Beans.get(InvoiceServiceProjectImpl.class).updateLines(invoice);
```

Comment:
Create a new interface InvoiceServiceProject that will be implemented by InvoiceServiceProjectImpl and inject it, in order to inject interface and not the implementation.